ngraph/frontend/onnx_import/src/op/dropout.cpp

```diff
@@ -0,0 +1,110 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <memory>
+
+#include "core/null_node.hpp"
+#include "default_opset.hpp"
+#include "exceptions.hpp"
+#include "ngraph/log.hpp"
+#include "ngraph/node.hpp"
+#include "op/dropout.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace
+            {
+                OutputVector build_dropout(const Node& node, bool training_mode)
+                {
+                    CHECK_VALID_NODE(
+                        node, !training_mode, "Training mode is not supported for Dropout op");
+
+                    const auto input_data = node.get_ng_inputs().at(0);
+                    const bool return_mask = node.get_outputs_size() > 1;
+
+                    if (return_mask)
+                    {
+                        const auto mask = std::make_shared<default_opset::Broadcast>(
+                            default_opset::Constant::create(
+                                ngraph::element::boolean, Shape{}, {true}),
+                            std::make_shared<default_opset::ShapeOf>(input_data));
+                        return {input_data, mask};
+                    }
+                    else
+                    {
+                        return {input_data};
+                    }
+                }
+            }
+
+            namespace set_12
+            {
+                OutputVector dropout(const Node& node)
+                {
+                    const auto ng_inputs = node.get_ng_inputs();
+                    // seed attribute and ratio input are ignored because traning mode is not
+                    // supported anyway
+                    bool training_mode = false; // default value
+                    if (ng_inputs.size() > 2)
+                    {
+                        if (!ngraph::op::is_null(ng_inputs.at(2)))
+                        {
+                            CHECK_VALID_NODE(
+                                node,
+                                ngraph::op::is_constant(ng_inputs.at(2).get_node_shared_ptr()),
+                                "Not constant (or omitted) training_mode input is not supported.");
+                            training_mode = as_type_ptr<default_opset::Constant>(
+                                                ng_inputs.at(2).get_node_shared_ptr())
+                                                ->cast_vector<bool>()[0];
```

Comment:
"Not constant ~(or omitted)~ training_mode input is not supported."
I believe with the latest changes we support ommited `training_mode` input with `false` as default.

Just wonder, maybe it's worth to change this `Constant` only restriction and allow provide models also with `Parameter` training_mode input and warning that it will be always treat as `false` , because at the current moment IE doesn't support training by definition. 
But here onnx importer rejects `Parameter` training_mode input only to check if there is `false` or `true`, while we can't perform `true` case anyway. 
In the compliance tests we have also `Parameter` training_mode input that will fail with such restriction.

Reply:
- description fixed
- I think that in real scenario if user pass `traning_mode` (instead of just omit it) training mode is rather expected so for me exception makes sense